Translation Uncontrolled components (#114)
reactjs/es.reactjs.org: content/docs/uncontrolled-components.md

}
```

[**Pruébalo en CodePen**](https://codepen.io/gaearon/pen/WooRWa?editors=0010)

> Ya que un componente no controlado mantiene la fuente de la verdad en el DOM, a veces es más fácil integrar el código React y el código no React cuando usas componentes no controlados. También puede ser un poco menos de código si quieres ser rápido y sucio. De lo contrario, normalmente debes utilizar componentes controlados.

Reviewer: ```suggestion
Ya que un componente es la fuente de la verdad en el DOM, a veces es más fácil integrar el código React y el código no React cuando usas componentes no controlados. También puede haber menos código si optas por una solución rápida y sin muchos miramientos. De lo contrario, deberías por lo general utilizar componentes controlados.
```

> Sí aún no tienes claro qué tipo de componente debes usar para una situación en particular, puedes encontrar [este articulo sobre entradas controladas y no controladas](http://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) para ser tu ayuda.

Reviewer: There is a typo in "Sí". Also, I would rephrase the last part to make it sound more idiomatic.
```suggestion
Si aún no tienes claro qué tipo de componente debes usar para una situación en particular, [este artículo sobre entradas controladas y no controladas](http://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) te puede resultar útil.
```

### Valores predeterminados

> En el ciclo de vida de renderizado React, el atributo `value` en los elementos del formulario reemplazará el valor en el DOM con un componente no controlado, a menudo quieres React para especificar el valor inicial, pero dejas las actualizaciones posteriores sin control. Para manejar este caso, puedes especificar un `defaultValue` en lugar de `value`.

Reviewer: ```suggestion
En el ciclo de vida de renderizado de React, el atributo `value` en los elementos del formulario reemplazará el valor en el DOM. Con un componente no controlado, a menudo quieres que React especifique el valor inicial, pero que las actualizaciones posteriores no estén controladas. Para manejar este caso, puedes especificar un atributo `defaultValue` en lugar de `value`.
```

}
```

> Del mismo modo, `<input type="checkbox">` y `<input type="radio">` respaldado por `defaultChecked`, y `<select>` y `<textarea>` respaldado por `defaultValue`.

Reviewer: ```suggestion
Del mismo modo, `<input type="checkbox">` e `<input type="radio">` admiten `defaultChecked`, y `<select>` y `<textarea>` admiten `defaultValue`.
```

## La etiqueta de entrada de archivo

> En HTML, un `<input type="file">` permite al usuario elegir uno o más archivos del almacenamiento en sus dispositivos para cargarlos a un servidor o manipularlos mediante JavaScript a través de   [File API](https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications).

Reviewer: ```suggestion
En HTML, un `<input type="file">` permite al usuario elegir uno o más archivos del almacenamiento en sus dispositivos para cargarlos a un servidor o manipularlos mediante JavaScript a través de la [API File](https://developer.mozilla.org/es/docs/Web/API/File/Using_files_from_web_applications).
```

```html
<input type="file" />
```

En React, un `<input type="file" />` siempre es un componente no controlado porque su valor solo puede ser establecido por un usuario, y no mediante programación.

> Debes utilizar File API para interactuar con ellos. El siguiente ejemplo muestra cómo crear un [referencia al nodo DOM](/docs/refs-and-the-dom.html) para acceder a los archivos en un controlador de envío:

Reviewer: ```suggestion
Debes utilizar la API File para interactuar con ellos. El siguiente ejemplo muestra cómo crear un [referencia al nodo DOM](/docs/refs-and-the-dom.html) para acceder a los archivos en un controlador de envío:
```
